### PR TITLE
Add support for KTX textures

### PR DESCRIPTION
**Pull request by CEksal**
This slightly reorganizes the `compressed_textures` part of the engine, and adds a ktx format loader.

### CODE REVIEW

File: lib/src/engine/compressed_texture_formats/ktx_texture.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  void _parseHeader() {
+    final bytes = ByteArray.fromBuffer(_buffer);
+    final magic = List.generate(12, (_) => bytes.readUnsignedByte());
+
+    for (final entry in magic.asMap().entries) {
+      if (entry.value != KtxFormat.ktx_magic[entry.key]) {
+        print('unrecognized magic, not a ktx header');
+        return;
```
Comment (mnordine) [inline lib/src/engine/compressed_texture_formats/ktx_texture.dart:26]: I think we should probably `throw` here.

File: lib/src/engine/compressed_texture_formats/ktx_texture.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    final endianness = bytes.readUnsignedInt();
+    switch (endianness) {
+      case KtxFormat.ktx_end_le:
+        // We're already reading in little endian
+        break;
+      case KtxFormat.ktx_end_be:
+        // Switch the ByteArray to reading in big endian
+        print('warning: ktx file is big endian, this is untested');
```
Comment (mnordine) [inline lib/src/engine/compressed_texture_formats/ktx_texture.dart:37]: Maybe add a TODO here.

File: lib/src/engine/compressed_texture_formats/ktx_texture.dart [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        bytes.endian = Endian.big;
+        break;
+      default:
+        print('unrecognized endianness, not a ktx header');
+        return;
```
Comment (mnordine) [inline lib/src/engine/compressed_texture_formats/ktx_texture.dart:42]: We should probably `throw` here.

File: lib/src/resources/texture_atlas_loader.dart [hunk 4]
```diff
@@ -89,6 +89,25 @@ class _TextureAtlasLoaderFile extends TextureAtlasLoader {
...
+    final filenameParts = filename.split('.');
+    var ext = filenameParts.last;
+
+    if (ext == 'gz') {
+      ext = filenameParts[filenameParts.length - 2];
+    }
+
+    CompressedTextureFileTypes type;
+
+    switch(ext) {
```
Comment (mnordine) [inline lib/src/resources/texture_atlas_loader.dart:101]:
```suggestion
    switch (ext) {
```

**Review by mnordine — CHANGES_REQUESTED** [review]

File: lib/src/engine/compressed_texture_formats/ktx_texture.dart (hunk 1 above)
Comment (CEksal) [reply lib/src/engine/compressed_texture_formats/ktx_texture.dart:26]: I thought so too, but I assumed there was a reason you made the PVR parser just print. I'll change both to throw.

File: lib/src/engine/compressed_texture_formats/ktx_texture.dart (hunk 2 above)
Comment (CEksal) [reply lib/src/engine/compressed_texture_formats/ktx_texture.dart:37]: I don't think we'll run into BE files, TexturePacker doesn't create them and neither does Compressonator. I'll mark a TODO to test those in the future though.

**Review by mnordine — APPROVED** [review]

**Pull request merged**